### PR TITLE
gettext: add a LangpackResolver path resolver

### PR DESCRIPTION
**Pull request by jhenstridge**
This is based on a suggestion from @didrocks.  As all of snapd, zsys and adsys implement a custom `PathResolver` to locate catalogs distributed via langpacks, it would make sense to provide a version of this resolver as part of go-gettext.  I'm leaving this marked as draft, since I'm not quite sure whether this is the right solution.

The `LangpackResolver` function is somewhat simpler than the ones in snapd or zsys, since it is not trying to check simplified versions of the provided locale: that's automatic now that #7 has been merged.  It also hard codes the `/usr/share/locale-langpack` directory, since that is where language pack catalogs will be located (if available), no matter what locale dir is configured for the domain.

This somewhat mirrors what OpenSUSE's langpack patch does:
   https://build.opensuse.org/package/view_file/Base:System/glibc/glibc-2.3.90-langpackdir.diff

But differs from the Ubuntu patch does, where the langpack catalogs are added as fallbacks, consulted after all the non-langpack catalogs have been checked:
   https://git.launchpad.net/ubuntu/+source/glibc/tree/debian/patches/ubuntu/local-altlocaledir.diff

The method used by the current Ubuntu patch would not fit into the existing PathResolver model.

### CODE REVIEW

File: gettext.go [hunk 1]
```diff
@@ -28,12 +27,26 @@ type translations struct {
...
 // DefaultResolver resolves paths in the standard format of:
 // <root>/<locale>/LC_MESSAGES/<domain>.mo
-func DefaultResolver(root string, locale string, domain string) string {
-	return path.Join(root, locale, "LC_MESSAGES", fmt.Sprintf("%s.mo", domain))
+func DefaultResolver(root, locale, domain string) string {
+	return path.Join(root, locale, "LC_MESSAGES", domain+".mo")
+}
+
+// LangpackResolver is a PathResolver that uses language pack catalogs
+// if available.
```
Comment (mvo5) [inline gettext.go:39]: Maybe expand where langpacks are used/come from? Something like:
```
...
//
// Language packs are mostly used by Ubuntu and it's derivatives, enabling this resolver will do no harm on other distros.
```

**Review by mvo5 — APPROVED** [review]
I think having this is fine, so :+1: for batteries included.

**Comment by jhenstridge** [comment]
My concern was more about whether this is the right way to handle langpacks.  It is close to what we've done in snapd and zsys, but differs from the glibc patch, where it looks like it adds both the normal and langpack catalogues to the search path.

That won't fit into the existing PathResolver API though, which is set up to return one path for each locale identifier.  So I'm wondering if some kind of `UseLangpacks` boolean flag on the TextDomain might be better?

**Comment by jhenstridge** [comment]
I'm closing this attempt in favour of PR #14.

**Pull request closed without merging**